openedx/features/course_experience/views/course_outline.py

```diff
@@ -23,15 +26,44 @@ def render_to_fragment(self, request, course_id=None, page_context=None, **kwarg
         """
         course_key = CourseKey.from_string(course_id)
         course_overview = get_course_overview_with_access(request.user, 'load', course_key, check_if_enrolled=True)
-
         course_block_tree = get_course_outline_block_tree(request, course_id)
+
         if not course_block_tree:
             return None
 
+        content_milestones = self.get_content_milestones(request, course_key)
+
         context = {
             'csrf': csrf(request)['csrf_token'],
             'course': course_overview,
             'blocks': course_block_tree,
+            'milestones': content_milestones
         }
         html = render_to_string('course_experience/course-outline-fragment.html', context)
         return Fragment(html)
+
+    def get_content_milestones(self, request, course_key):
+        """
+        Returns dict of subsections with prerequisites and whether the prerequisite has been completed or not
+        """
+        course_content_milestones = {}
+
+        all_course_prereqs = get_course_content_milestones(
+            course_id=course_key,
+            content_id=None,
+            relationship='requires',
+            user_id=None)
```

Comment:
BTW, if you go with option 2 on the method signature change, this will just become:
```
all_course_prereqs = get_course_content_milestones(course_id=course_key)
```